src/backendLogic.js

```diff
@@ -38,26 +45,102 @@ app.get('/', async (req, res) => {
-        if (userList[i].username === req.body.username) {
-            isValidUser = true;
-            if (userList[i].password === req.body.password) {
-                res.send(JSON.stringify("Login Success"));
-            } else {
-                res.statusCode = 401;
-                res.send(JSON.stringify("Wrong Password"));
-            }
-        }
-    }
-    if (!isValidUser) {
-        res.statusCode = 400;
-        res.send(JSON.stringify("Invalid User"));
+    const username = req.body.username;
+    const password = req.body.password;
+
+    switch (await checkLoginInfo(username, password)) {
+        case LOGIN_SUCCESS:
+            res.send(JSON.stringify(LOGIN_SUCCESS));
+            break;
+        case WRONG_PASSWORD:
+            res.statusCode = 401;
+            res.send(JSON.stringify(WRONG_PASSWORD));
+            break;
+        case INVALID_USER:
+            res.statusCode = 400;
+            res.send(JSON.stringify(INVALID_USER));
+            break;
     }
+});
 
+//For signing up
+app.post('/sign_up', async (req, res) => {
+    const username = req.body.username;
+    const password = req.body.password;
 
+    switch (await checkSignUpInfo(username)) {
+        case SIGN_UP_SUCCESS:
+            //Add the new user to the json
+            await addNewUserToDB(username, password);
+            console.log("sign up success");
+            res.send(JSON.stringify(SIGN_UP_SUCCESS));
+            break;
+        case ALREADY_USED_ID:
+            console.log("already used id");
+            res.statusCode = 402;
+            res.send(JSON.stringify(ALREADY_USED_ID));
+            break;
+    }
 });
 
 app.listen(port, () => {
     console.log(`Server running on port ${port}`);
-});
\ No newline at end of file
+});
+
+async function getFileContent(filePath) {
+    let fileContent;
+    try {
+        fileContent = await readFileAsync(filePath, 'utf-8');
+        return fileContent;
+    } catch (err) {
+        console.log('error: ', err);
+    }
+}
+
+async function checkLoginInfo(username, password) {
+    return await (async () => {
+        try{
+            const sql = `SELECT password from users WHERE username=?`;
+            const queryResult = await queryDBAsync(sql, username);
+
+            if(queryResult.length === 0){
+                return INVALID_USER;
+            } else if(queryResult[0].password === password){
+                return LOGIN_SUCCESS;
+            } else {
+                return WRONG_PASSWORD;
+            }
+        } catch (err) {
+            console.log("checkLoginInfo err"+err);
+        }
+    })();
+}
+
+async function checkSignUpInfo(username) {
+    return await (async () => {
+        try{
+            const sql = `SELECT * from users WHERE username=?`;
+            const queryResult = await queryDBAsync(sql, username);
+            if (queryResult.length > 0) {
+                return ALREADY_USED_ID;
+            } else {
+                return SIGN_UP_SUCCESS;
```

Comment:
Looks like we are checking if this user is in the DB or not. Let's make a separate keyword for this.

Reply:
done